### ISSUE TITLE
80% on security test

### ISSUE THREAD

**Issue by a1300** [GNYIO/gny-wallet#55]
When looking at https://sitecheck.dev/?url=https://wallet.gny.io

![image](https://user-images.githubusercontent.com/11912239/103466697-6eb05a00-4d47-11eb-8e9d-0782dd113477.png)
![image](https://user-images.githubusercontent.com/11912239/103466703-78d25880-4d47-11eb-8202-ce4d200b9a67.png)
![image](https://user-images.githubusercontent.com/11912239/103466706-825bc080-4d47-11eb-9f89-c8e1c8ff9bc6.png)

**Comment by a1300** [comment]
@xpgeng 100% now :slightly_smiling_face: :+1: 

![image](https://user-images.githubusercontent.com/11912239/103747845-8ddb1000-5003-11eb-8956-91f819d67ad9.png)

Fixed with PR: #57

**Comment by a1300** [comment]
@xpgeng reopened this ticket

![image](https://user-images.githubusercontent.com/11912239/104030252-62a31d00-51cb-11eb-8bec-2c4442893a7b.png)

**Comment by a1300** [comment]
I reset the commit. You can test the http header with the netlify preview of your PR

```bash
git reset --hard 0ac8867c702b1b2122193dc956e2ef0dbf36bec1
```

**Comment by a1300** [comment]
@xpgeng the wallet is only loading from `wallet.gny.io` and `testnet.gny.io`. So from the same domain, but different subdomains.
![image](https://user-images.githubusercontent.com/11912239/104035983-22479d00-51d3-11eb-826a-424c51d59639.png)


I think we need to do: 
```
default-src 'self' testnet.gny.io
# self ´= wallet.gny.io
```
or
```
default-src 'self' *.gny.io
```

**Comment by a1300** [comment]
Only allowing `testnet.gny.io` instead of `*.gny.io` is more strict and will prevent accidentally request the wrong network. We followed the [netlify docs](https://docs.netlify.com/configure-builds/file-based-configuration/#inject-environment-variable-values) to change the `public/_headers` file while building the project on netlify.

Our environment variables on netlify:
![image](https://user-images.githubusercontent.com/11912239/104084060-2b248700-5244-11eb-9eff-fd6866de09ce.png)


Build command **before and after** (on netlify):
```diff
- npm run build
+ sed -i s/GNY_API_ENDPOINT_PLACEHOLDER/${VUE_APP_GNY_ENDPOINT}/g public/_headers && npm run build
```

https://github.com/GNYIO/gny-wallet/blob/e15b22b47d9633472acebae717ec4f811a7ec8d6/public/_headers#L1-L6

This **changes** the `GNY_API_ENDPOINT_PLACEHOLDER` string to the `$VUE_APP_GNY_ENDPOINT` environment variable. This can be also used then for the **mainnet**

**Comment by a1300** [comment]
Closed with PR: #59
